core/templates/dev/head/domain/topic/TopicObjectFactory.js

```diff
@@ -21,15 +21,18 @@ oppia.factory('TopicObjectFactory', ['SubtopicObjectFactory',
   function(SubtopicObjectFactory) {
     var Topic = function(
         id, name, description, languageCode, canonicalStoryIds,
-        additionalStoryIds, uncategorizedSkillIds, nextSubtopicId, version,
-        subtopics) {
+        additionalStoryIds, uncategorizedSkillIds,
+        uncategorizedSkillDescriptions, nextSubtopicId, version, subtopics) {
       this._id = id;
       this._name = name;
       this._description = description;
       this._languageCode = languageCode;
       this._canonicalStoryIds = canonicalStoryIds;
       this._additionalStoryIds = additionalStoryIds;
-      this._uncategorizedSkillIds = uncategorizedSkillIds;
+      this._uncategorizedSkills = {
```

Comment:
Seems like odd modelling. Should be a list of dicts (or even better, a list of Skill or SkillSummary domain objects? Or whatever the conceptual representation for "id + description" is)

Reply:
Done, changed it to a list of dicts.